okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientStream.java

```diff
@@ -96,6 +96,10 @@ public void transportHeadersReceived(List<Header> headers, boolean endOfStream)
     synchronized (lock) {
       if (endOfStream) {
         transportTrailersReceived(Utils.convertTrailers(headers));
+        if (getTransportError() != null) {
```

Comment:
I don't understand why the transport error has an impact on whether we should call transport.finishStream(id(), null).